File: rest-service/src/main/java/com/example/restservice/service/ImageService.java
package com.example.restservice.service;

import com.example.restservice.model.Image;

import java.util.List;

/**
 * Interfața care definește operațiile disponibile pentru gestionarea imaginilor în sistem.
 * Aceste operații includ obținerea tuturor imaginilor, obținerea unei imagini după ID,
 * crearea, actualizarea și ștergerea imaginilor.
 */
public interface ImageService {
    /**
     * Obține o listă cu toate imaginile din sistem.
     *
     * @return Lista cu imagini
     */
    List<Image> getAllImages();

    /**
     * Obține o imagine după ID-ul specificat.
     *
     * @param id ID-ul imaginii căutate
     * @return Imaginea cu ID-ul specificat sau null dacă nu există
     */
    Image getImageById(Long id);

    /**
     * Creează o imagine nouă în sistem.
     *
     * @param image Imaginea care urmează să fie creată
     * @return Imaginea creată
     */
    Image createImage(Image image);

    /**
     * Actualizează o imagine existentă în sistem.
     *
     * @param id          ID-ul imaginii care urmează să fie actualizată
     * @param updateImage Imaginea actualizată
     * @return Imaginea actualizată sau null dacă nu există o imagine cu ID-ul specificat
     */
    Image updateImage(Long id, Image updateImage);

    /**
     * Șterge o imagine existentă din sistem.
     *
     * @param id ID-ul imaginii care urmează să fie ștearsă
     */
    void deleteImage(Long id);

    /**
     * Obtine toate imaginile unei stiri
     * @param newsID id-ul stirii cautate
     * @return lista de imagini
     */
    List<Image> getAllImagesByNewsID(Long newsID);
}

File: front-site/src/app/service/image.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { News } from './news.service';

// Definirea interfeței Image
export interface Image {
  imageId: number;
  news: News;
  url: string;
}

@Injectable({
  providedIn: 'root'
})
export class ImageService {
  private baseUrl = 'http://localhost:8080/images';

  constructor(private http: HttpClient) { }

  // Metodă pentru obținerea tuturor imaginilor
  getAllImages(): Observable<Image[]> {
    return this.http.get<Image[]>(this.baseUrl);
  }

  // Metodă pentru obținerea unei imagini după ID
  getImageById(id: number): Observable<Image> {
    return this.http.get<Image>(`${this.baseUrl}/${id}`);
  }

  // Metodă pentru crearea unei imagini noi
  createImage(image: Image): Observable<Image> {
    return this.http.post<Image>(this.baseUrl, image);
  }

  // Metodă pentru actualizarea unei imagini existente
  updateImage(id: number, image: Image): Observable<Image> {
    return this.http.put<Image>(`${this.baseUrl}/${id}`, image);
  }

  // Metodă pentru ștergerea unei imagini după ID
  deleteImage(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }

  // Metodă pentru obținerea imaginilor după ID-ul știrii
  getImagesByNewsId(id: number): Observable<Image[]> {
    return this.http.get<Image[]>(`${this.baseUrl}/getImagesByNewsId/${id}`);
  }
}
